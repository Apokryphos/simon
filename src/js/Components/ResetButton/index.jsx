import React from 'react';
import PropTypes from 'prop-types';

function ResetButton(props) {
  return (
    <div className="button-container" title="Resets the game.">
      <button className="reset-button" onClick={props.onClick} />
      <span className="button-label">Reset</span>
    </div>
  );
}

ResetButton.propTypes = {
  onClick: PropTypes.func.isRequired,
};

export default ResetButton;
