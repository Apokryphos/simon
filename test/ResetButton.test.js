import React from 'react';
import { shallow } from 'enzyme';
import ResetButton from 'Components/ResetButton';

describe('ResetButton rendering', () => {
  it('Renders OK', () => {
    const onClick = () => {};

    const wrapper = shallow(<ResetButton onClick={onClick}/>);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.props('onClick') === onClick);
  });
});
