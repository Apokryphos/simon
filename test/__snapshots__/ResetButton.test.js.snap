// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetButton rendering Renders OK 1`] = `
<div
  className="button-container"
  title="Resets the game."
>
  <button
    className="reset-button"
    onClick={[Function]}
  />
  <span
    className="button-label"
  >
    Reset
  </span>
</div>
`;
